src/Parsers/ASTAlterQuery.cpp

```diff
@@ -569,58 +573,51 @@ ASTPtr ASTAlterQuery::clone() const
     return res;
 }
 
-void ASTAlterQuery::formatQueryImpl(const FormatSettings & settings, FormatState & state, FormatStateStacked frame) const
+void ASTAlterQuery::formatQueryImpl(FormattingBuffer out) const
 {
-    frame.need_parens = false;
+    out.setNeedParens(false);
 
-    std::string indent_str = settings.one_line ? "" : std::string(4u * frame.indent, ' ');
-    settings.ostr << (settings.hilite ? hilite_keyword : "") << indent_str;
+    out.writeIndent();
 
     switch (alter_object)
     {
         case AlterObjectType::TABLE:
-            settings.ostr << "ALTER TABLE ";
+            out.writeKeyword("ALTER TABLE ");
             break;
         case AlterObjectType::DATABASE:
-            settings.ostr << "ALTER DATABASE ";
+            out.writeKeyword("ALTER DATABASE ");
             break;
         case AlterObjectType::LIVE_VIEW:
-            settings.ostr << "ALTER LIVE VIEW ";
+            out.writeKeyword("ALTER LIVE VIEW ");
             break;
         default:
             break;
     }
 
-    settings.ostr << (settings.hilite ? hilite_none : "");
-
     if (table)
     {
         if (database)
         {
-            settings.ostr << indent_str << backQuoteIfNeed(getDatabase());
-            settings.ostr << ".";
+            out.writeIndent();
+            out.ostr << backQuoteIfNeed(getDatabase());
+            out.ostr << ".";
         }
-        settings.ostr << indent_str << backQuoteIfNeed(getTable());
+        out.writeIndent();
+        out.ostr << backQuoteIfNeed(getTable());
     }
     else if (alter_object == AlterObjectType::DATABASE && database)
     {
-        settings.ostr << indent_str << backQuoteIfNeed(getDatabase());
+        out.writeIndent();
+        out.ostr << backQuoteIfNeed(getDatabase());
     }
 
-    formatOnCluster(settings);
+    formatOnCluster(out);
 
-    FormatStateStacked frame_nested = frame;
-    frame_nested.need_parens = false;
-    if (settings.one_line)
-    {
-        frame_nested.expression_list_prepend_whitespace = true;
-        command_list->formatImpl(settings, state, frame_nested);
-    }
-    else
-    {
-        frame_nested.expression_list_always_start_on_new_line = true;
-        command_list->as<ASTExpressionList &>().formatImplMultiline(settings, state, frame_nested);
-    }
+    FormattingBuffer copy = out.copy().setNeedParens(false);
```

Comment:
It seems you don't need to copy `out` here because earlier in this function you already set `NeedParens` to false and also because this function owns `out` and can change it.